static/js/components/tiles/chart_action_icons.tsx

```diff
@@ -0,0 +1,171 @@
+import { UncontrolledTooltip } from "reactstrap";
+
+import {
+  GA_EVENT_TILE_DOWNLOAD,
+  GA_EVENT_TILE_EXPLORE_MORE,
+  GA_PARAM_TILE_TYPE,
+  triggerGAEvent,
+} from "../../shared/ga_events";
+import { ChartDownload, ChartDownloadSpec } from "./modal/chart_download";
+
+/** Component for a single action icon */
+
+interface ActionIconPropType {
+  // Container element the tooltip should attach to
+  container?: HTMLElement;
+  // Name of "Material Icons Outlined" icon to show
+  icon: string;
+  // Callback to run when icon is clicked
+  onClickHandler: (event: React.MouseEvent<HTMLElement>) => void;
+  // What to display in a tooltip on hover
+  tooltipContent: string | JSX.Element;
+  // If icon is being used as a link, the url to go to
+  url?: string;
+}
+
+function ActionIcon(props: ActionIconPropType): JSX.Element {
+  const linkRef = useRef<HTMLAnchorElement>();
+  const [linkIsReady, setLinkIsReady] = useState(false);
+
+  // Check if anchor link has rendered.
+  // Tooltip needs the anchor link to render first to have a target to bind to.
+  useEffect(() => {
+    if (linkRef.current) {
+      setLinkIsReady(true);
+    }
+  }, [linkRef]);
+
+  return (
+    <div className="outlink-item action-icon">
+      <a
+        href={props.url || "#"}
+        onClick={props.onClickHandler}
+        ref={linkRef}
+        rel={props.url ? "" : "noopener noreferrer"}
+        target={props.url ? "" : "_blank"}
+      >
+        <span className="material-icons-outlined">{props.icon}</span>
+      </a>
+      {linkIsReady && (
+        <UncontrolledTooltip
+          boundariesElement="window"
+          container={props.container}
+          delay={200}
+          placement="top"
+          target={linkRef.current}
+          trigger="hover"
+        >
+          {props.tooltipContent}
+        </UncontrolledTooltip>
+      )}
+    </div>
+  );
+}
+
+/** Component for a row of icons to place in chart footer */
+
+interface ChartActionsPropType {
+  // Containing HTML element to attach tooltips/modals to
+  container?: HTMLElement;
+  // Callback to get specs of the chart to download if "download" is clicked.
+  getChartDownloadSpec?: () => ChartDownloadSpec;
+  // Id of the chart to provide feedback for
+  id: string;
+  // Link to explore more. Only show explore button if this object is non-empty.
+  exploreLink?: { displayText: string; url: string };
+}
+
+export function ChartActions(props: ChartActionsPropType): JSX.Element {
+  const [chartDownloadSpec, setChartDownloadSpec] =
+    useState<ChartDownloadSpec>(null);
+  const [showDownloadModal, setShowDownloadModal] = useState(false);
+
+  return (
+    <>
+      <div className="outlinks">
+        {props.getChartDownloadSpec && (
+          <ActionIcon
+            container={props.container}
+            icon="download"
+            onClickHandler={(event) => {
+              event.preventDefault();
+              triggerGAEvent(GA_EVENT_TILE_DOWNLOAD, {
```

Comment:
Could the page navigation complete before the ga event?
https://stackoverflow.com/a/50062249